libs/clients/documents-v2/src/lib/documentsClientV2.service.ts

```diff
@@ -0,0 +1,179 @@
+import {
+  CustomersApi,
+  CustomersListDocumentsOrderEnum,
+  CustomersListDocumentsSortByEnum,
+  CustomersWantsPaperMailRequest,
+} from '../../gen/fetch'
+import { Injectable } from '@nestjs/common'
+import { DocumentDto, mapToDocument } from './dto/document.dto'
+import { ListDocumentsInputDto } from './dto/listDocuments.input'
+import { ListDocumentsDto } from './dto/documentList.dto'
+import { isDefined } from '@island.is/shared/utils'
+import { mapToDocumentInfoDto } from './dto/documentInfo.dto'
+import { MailAction } from './dto/mailAction.dto'
+
+@Injectable()
+export class DocumentsClientV2Service {
+  constructor(private api: CustomersApi) {}
+
+  async getDocumentList(
+    input: ListDocumentsInputDto,
+  ): Promise<ListDocumentsDto | null> {
+    const documents = await this.api.customersListDocuments({
+      ...input,
+      kennitala: input.nationalId,
+      senderKennitala: input.senderNationalId,
+      order: input.order
+        ? CustomersListDocumentsOrderEnum[input.order]
+        : undefined,
+      sortBy: input.sortBy
+        ? CustomersListDocumentsSortByEnum[input.sortBy]
+        : undefined,
+    })
+
+    if (!documents.totalCount) {
+      //throw maybe?
+      return null
```

Comment:
I think returning null would be ideal instead of throwing here.

Reply:
ok so leave it as is

Reply:
~ Then remove the comment?

Reply:
will do